src/client/datascience/raw-kernel/rawJupyterSession.ts

```diff
@@ -63,33 +64,27 @@ export class RawJupyterSession extends BaseJupyterSession {
 
     // Connect to the given kernelspec, which should already have ipykernel installed into its interpreter
     public async connect(
-        resource: Resource,
-        kernelConnection: KernelConnectionMetadata,
-        timeout: number,
         cancelToken?: CancellationToken,
         disableUI?: boolean
     ): Promise<KernelConnectionMetadata | undefined> {
         // Save the resource that we connect with
         let newSession: RawSession;
-        trackKernelResourceInformation(resource, { kernelConnection });
+        trackKernelResourceInformation(this.resource, { kernelConnection: this.kernelConnectionMetadata });
         const stopWatch = new StopWatch();
         try {
             // Try to start up our raw session, allow for cancellation or timeout
             // Notebook Provider level will handle the thrown error
-            newSession = await this.startRawSession(resource, kernelConnection, timeout, cancelToken, disableUI);
+            newSession = await this.startRawSession(cancelToken, disableUI);
 
             // Only connect our session if we didn't cancel or timeout
-            sendKernelTelemetryEvent(resource, Telemetry.RawKernelSessionStartSuccess);
-            sendKernelTelemetryEvent(resource, Telemetry.RawKernelSessionStart, stopWatch.elapsedTime);
+            sendKernelTelemetryEvent(this.resource, Telemetry.RawKernelSessionStartSuccess);
+            sendKernelTelemetryEvent(this.resource, Telemetry.RawKernelSessionStart, stopWatch.elapsedTime);
             traceInfo('Raw session started and connected');
             this.setSession(newSession);
 
             // Listen for session status changes
             this.session?.statusChanged.connect(this.statusHandler); // NOSONAR
 
-            // Update kernelspec and interpreter
-            this.kernelConnectionMetadata = newSession.kernelProcess?.kernelConnectionMetadata;
```

Comment:
This is now gone, this is no longer required, in the ctor we define this property, hence its never nullable.
I.e. we always have the right value when the class is instantiated